src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -281,21 +319,303 @@ bool UnixNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
 
     *ppPreviousTransitionFrame = NULL;
 
-    if (!VirtualUnwind(pRegisterSet))
+    if (!VirtualUnwind(pRegisterSet)) 
     {
         return false;
     }
 
     return true;
 }
 
+bool UnixNativeCodeManager::IsUnwindable(PTR_VOID pvAddress)
+{
+    // VirtualUnwind can't unwind prologues.
+    return TrailingEpilogueInstructionsCount(pvAddress) == 0;
+}
+
+#define SIZE64_PREFIX 0x48
+#define ADD_IMM8_OP 0x83
+#define ADD_IMM32_OP 0x81
+#define JMP_IMM8_OP 0xeb
+#define JMP_IMM32_OP 0xe9
+#define JMP_IND_OP 0xff
+#define LEA_OP 0x8d
+#define REPNE_PREFIX 0xf2
+#define REP_PREFIX 0xf3
+#define POP_OP 0x58
+#define RET_OP 0xc3
+#define RET_OP_2 0xc2
+
+#define IS_REX_PREFIX(x) (((x) & 0xf0) == 0x40)
+
+// when stopped in an epilogue, returns the count of remaining stack-consuming instructions
+// otherwise returns 0
+int UnixNativeCodeManager::TrailingEpilogueInstructionsCount(PTR_VOID pvAddress)
+{
+#ifdef TARGET_AMD64
+
+    //
+    // Everything below is inspired by the code in minkernel\ntos\rtl\amd64\exdsptch.c file from Windows
+    // For details see similar code in OOPStackUnwinderAMD64::UnwindEpilogue
+    //
+    //
+    //    
+    // A canonical epilogue sequence consists of the following operations:
+    //
+    // 1. Optional cleanup of fixed and dynamic stack allocations, which is
+    //    considered to be outside of the epilogue region.
+    //
+    //    add rsp, imm
+    //        or
+    //    lea rsp, disp[fp]
+    //
+    // 2. Zero or more pop nonvolatile-integer-register[0..15] instructions.
+    //
+    //    pop r64
+    //        or
+    //    REX.R pop r64
+    //
+    // 3. An optional one-byte pop r64 to a volatile register to clean up an
+    //    RFLAGS register pushed with pushfq.
+    //
+    //    pop rcx
+    //
+    // 4. A control transfer instruction (ret or jump, in a case of a tailcall)
+    //    For the purpose of inferring the state of the stack, ret and jump can be
+    //    considered the same.
+    //
+    //    ret 0
+    //        or
+    //    jmp imm
+    //        or
+    //    jmp [target]
+    //
+
+    // if we are in an epilogue, there will be at least one instruction left.
+    int trailingEpilogueInstructions = 1;
+    uint8_t* NextByte = (uint8_t*)pvAddress;
```

Comment:
A nit - can you please fix the naming convention of some of the locals (start with lowercase)?